Add beginning of garden lattice chapter

d33bs opened the pull request:
<!-- _modified from [EmbeddedArtistry](https://embeddedartistry.com/blog/2017/08/04/a-github-pull-request-template-for-your-projects/)_
_referenced with modifications from [pycytominer](https://github.com/cytomining/pycytominer/blob/master/.github/PULL_REQUEST_TEMPLATE.md)_ -->

# Description

Adds beginning of garden lattice chapter with an introduction to help set the themes. As part of this I had to make several adjustments to linting to help things align.

References #23 

## What is the nature of your change?

- [x] Content additions or updates (adds or updates content)
- [ ] Bug fix (fixes an issue).
- [x] Enhancement (adds functionality).
- [ ] Breaking change (these changes would cause existing functionality to not work as expected).

# Checklist

Please ensure that all boxes are checked before indicating that this pull request is ready for review.

- [x] I have read the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own contributions.
- [x] I have commented my content, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to related documentation (outside of book content).
- [x] My changes generate no new warnings.
- [x] New and existing tests pass locally with my changes.
- [ ] I have added tests that prove my additions are effective or that my feature works.
- [x] I have deleted all non-relevant text in this pull request template.

gwaybio commented on src/almanac/garden-lattice/garden-lattice.md:
```diff
@@ -0,0 +1,32 @@
+
+Lattices, reminiscent of living structures built with willow or bamboo, emerge as a weaved foundation, both flexible and systematically resilient, protecting and inspiring new growth.
+"The willow submits to the wind and prospers until one day it is many willows - a wall against the wind." {cite:p}`herbert_dune_2010`.
+These lattice also signify our collective friendship and interdependence. They echo the sentiment of "Be like bamboo. The higher you grow, the deeper you bow," (a Chinese proverb) embodying strength through flexibility and unity.
```
```suggestion
These lattice also signify our collective friendship and interdependence.
They echo the sentiment of "Be like bamboo. The higher you grow, the deeper you bow," (a Chinese proverb) embodying strength through flexibility and unity.
```

d33bs replied:
Thanks for this!

gwaybio commented on src/almanac/garden-lattice/garden-lattice.md:
```diff
@@ -0,0 +1,32 @@
+The lattice also guards against both direct and indirect challenges we face on our journey as software gardeners.
+These encumbrances can "overheat" our growth potential, leading us to early burnout or burn-down of gardens.
+The lattice helps us as a carbon sequester, mitigating the excess heat caused by an imbalance in emissions, similar to bamboo garden forests {cite:p}`nath_managing_2015`.
+Our individual experiences and personal network consume these difficulties to further grow the lattice through acts of harmonization.
```
What is an act of harmonization? Maybe provide an example?

d33bs replied:
Thanks, great question, will follow up with an example here.

gwaybio commented on src/almanac/garden-lattice/garden-lattice.md:
```diff
@@ -0,0 +1,32 @@
+# Garden Lattice
```
Is there an example image we can use?

d33bs replied:
Thanks, added an image here.

falquaddoomi commented on src/almanac/garden-lattice/garden-lattice.md:
```diff
@@ -0,0 +1,32 @@
+
+Lattices, reminiscent of living structures built with willow or bamboo, emerge as a weaved foundation, both flexible and systematically resilient, protecting and inspiring new growth.
+"The willow submits to the wind and prospers until one day it is many willows - a wall against the wind." {cite:p}`herbert_dune_2010`.
+These lattice also signify our collective friendship and interdependence. They echo the sentiment of "Be like bamboo. The higher you grow, the deeper you bow," (a Chinese proverb) embodying strength through flexibility and unity.
```
```suggestion
These lattices also signify our collective friendship and interdependence. They echo the sentiment of "Be like bamboo. The higher you grow, the deeper you bow," (a Chinese proverb) embodying strength through flexibility and unity.
```

d33bs replied:
Great catch, thanks!

falquaddoomi commented on src/almanac/garden-lattice/garden-lattice.md:
```diff
@@ -0,0 +1,32 @@
+> "To plant a garden is to believe in tomorrow."
+> \- Audrey Hepburn
+
+We enter the book through a garden lattice, an echo and forward reverberation of people who cultivate software.
```
I'm not sure I understand what you mean by "echo and forward reverberation". What's the difference here between an echo and a reverberation? Perhaps reducing it to either "echo" or "reverberation" would be easier to understand, unless you're drawing an important distinction between them.

d33bs replied:
Thanks for this, I think I felt that "echo" was past-tense, and "reverberation" was future-facing. Mostly I wanted to channel that this isn't a section stuck in the past. I think of a better way to state this and revise.

falquaddoomi commented on src/almanac/garden-lattice/garden-lattice.md:
I just want to say I like the metaphor of software as a lattice. I can definitely see how software grows together and inherits strength from the whole.

I don't know if you're planning to get to this later on, but I suppose it also suffers maladies as a whole, e.g. when some foundational piece breaks, or when a previously friendly collaborator that formed a part of the lattice becomes hostile. Just my two cents, take it as you will.

d33bs replied:
Thanks! This is just an introduction deeper sections where I'd like to address topics like you mention.

> I don't know if you're planning to get to this later on, but I suppose it also suffers maladies as a whole, e.g. when some foundational piece breaks, or when a previously friendly collaborator that formed a part of the lattice becomes hostile. Just my two cents, take it as you will.

There are definitely anti-patterns to growth which need to be addressed, maybe in the form of "positive and negative testing" scenarios (things we want more of vs things we want to limit or remove). These stem from foundational (death, geographic disconnection) to more complex like miscommunication or dishonesty. Generally my feeling at the moment is that the lattice is formed of both personal and interpersonal experiences, and that these experiences are woven together in a way to form structure where otherwise there is none. The structure has to be resilient enough to withstand the elements, parasites, overgrowth, disconnection (where the lattice is no longer forming). From the Almanac's perspective, we should catalog these patterns where possible for awareness and to provide tools for growth despite (or perhaps sometimes because of) destruction.

d33bs replied:
Added https://github.com/software-gardening/almanac/issues/35 to help address this moving forward.

falquaddoomi commented on src/almanac/garden-lattice/garden-lattice.md:
```diff
@@ -0,0 +1,32 @@
+These lattice also signify our collective friendship and interdependence. They echo the sentiment of "Be like bamboo. The higher you grow, the deeper you bow," (a Chinese proverb) embodying strength through flexibility and unity.
+This concept mirrors the organic growth seen in software gardens, where a system of interweaved latticework supports the development of intricate and resilient structures.
+
+The lattice also guards against both direct and indirect challenges we face on our journey as software gardeners.
```
I'm curious about this idea, and like other comments mentioned, I think some examples drawn from the experiences of everyday software developers would help make things more concrete. If you're planning to paint broad metaphors first and then pin them down later (e.g., in another section) with examples, that would be fine with me.

d33bs replied:
Thanks - definitely this is a bit abstract at the moment and I plan to go deeper with later portions of this chapter.

falquaddoomi commented on styles/config/vocabularies/almanac/accept.txt:
Kind of a shame the linter can't detect quotes and leave them as-is. I wonder if there's an invisible "no QA" annotation you could add rather than having to whitelist more and more terms as the almanac grows.

d33bs replied:
Thanks, good point, I believe there's an ignore rule for this through an HTML comment. I'll make an update here for this.

d33bs replied:
I found I could do this for `codespell` but not for `vale.Spelling`. I ended up adding a configuration rule to the `pyproject.toml` which is able to help with this.

falquaddoomi reviewed (commented):
Thought-provoking stuff. Altogether it sounds good; I left a few questions and a minor typo fix.

d33bs commented:
Thank you @gwaybio and @falquaddoomi for your reviews! I've addressed all comments and will merge this in now. Note: the Netlify previews appear to be having an issue which is unrelated to the changes in code (Python location within Netlify environment).
